core_test/carmirror_test.go

```diff
@@ -208,10 +217,25 @@ func MockBatchTransfer(sender_store *mock.Store, receiver_store *mock.Store, roo
 		latency_ms,
 	}
 
+	// If diagram is true, wrap the orchestrators in a diagrammer, to generate mermaid diagrams.
+	var batchSourceOrchestratorDiagrammer *diagrammer.StateDiagrammer
+	var batchSinkOrchestratorDiagrammer *diagrammer.StateDiagrammer
+	var batchSourceOrchestrator Orchestrator[BatchState]
+	var batchSinkOrchestrator Orchestrator[BatchState]
+	if diagram {
+		batchSourceOrchestratorDiagrammer = diagrammer.NewStateDiagrammer("BatchSourceOrchestrator")
+		batchSinkOrchestratorDiagrammer = diagrammer.NewStateDiagrammer("BatchSinkOrchestrator")
+		batchSourceOrchestrator = diagrammed.NewDiagrammedBatchSourceOrchestrator(NewBatchSourceOrchestrator(), batchSourceOrchestratorDiagrammer)
```

Comment:
Just a thought.... MockBatchTransfer is already a bit long/hard to follow for my taste. Could we hide all of this in DiagrammedBatchSourceOrchestrator? You could make Notify for BEGIN_SESSION create all the state you need and then have END_SESSION write the file. That way all that MockBatchTransfer would only need a couple of extra lines in it.